pkg/predictor_source/inferenceservice_registry.go

```diff
@@ -68,6 +70,54 @@ func BuildBasePredictorFromInferenceService(isvc *v1beta1.InferenceService) (*v1
 	return p, nil
 }
 
+// Return secretKey, bucket, modelPath, schemaPath, and error
+func processInferenceServiceStorage(inferenceService *v1beta1.InferenceService, nname types.NamespacedName) (
+	secretKey, bucket, modelPath, schemaPath string, err error) {
+	secretKey = inferenceService.ObjectMeta.Annotations[v1beta1.SecretKeyAnnotation]
+	_, frameworkSpec := inferenceService.Spec.Predictor.GetPredictorFramework()
+	storageUri := frameworkSpec.StorageURI
+	storageSpec := frameworkSpec.Storage
+	if storageUri == nil {
+		if storageSpec == nil || storageSpec.Path == nil {
+			return "", "", "", "", fmt.Errorf("the InferenceService %v must have either the storageUri or the storage.path", nname)
+		}
+		modelPath = *storageSpec.Path
+		if storageSpec.SchemaPath != nil {
+			schemaPath = *storageSpec.SchemaPath
+		}
+	} else {
+		if storageSpec != nil && storageSpec.Path != nil {
+			return "", "", "", "", fmt.Errorf("the InferenceService %v cannot have both the storageUri and the storage.path", nname)
+		}
+		if !strings.HasPrefix(*storageUri, "s3://") {
+			return "", "", "", "", nil
+		}
+		u, err := url.Parse(*storageUri)
+		if err != nil {
+			return "", "", "", "", err
+		}
```

Comment:
@Tomcli kind of a nit but may as well simplify like this:
```suggestion
		u, err := url.Parse(*storageUri)
		if err != nil || u.Scheme != "s3" {
			return "", "", "", "", err
		}
```